### PR TITLE
Provide Support for Multi-Cloud AK/SK Management

### PR DESCRIPTION
**Pull request by himanshuvar**
Support keystone credential retrieval mechanism and provide signing and authenticating requests to manage multi-cloud APIs.

### CODE REVIEW

File: api/pkg/filters/signature/credentials/credential.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2018 Huawei Technologies Co., Ltd. All Rights Reserved.
```
Comment (leonwanghui) [inline api/pkg/filters/signature/credentials/credential.go]: Please change copyright time to 2019

File: api/pkg/filters/signature/credentials/credential_test.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
+package credentials
```
Comment (leonwanghui) [inline api/pkg/filters/signature/credentials/credential_test.go]: Please add the license header.

File: api/pkg/filters/signature/credentials/keystonecredentials/provider.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2018 Huawei Technologies Co., Ltd. All Rights Reserved.
```
Comment (leonwanghui) [inline api/pkg/filters/signature/credentials/keystonecredentials/provider.go]: Same here.

**Comment by himanshuvar** [comment]
@wisererik @leonwanghui

**Pull request merged**